README.md

```diff
@@ -2,43 +2,44 @@
 
 This module provides the required Azure resources to integrate an Azure subscription with Uptycs.
 
+## Prerequisites [Do Not Skip]
 
-It creates the following resources:
+Ensure you have the following privileges before you execute the Terraform Script:
 
-* Application
-* Service principal to the application
-* It attaches the following roles and permissions to the service principal:
+- Administrative roles:
+  - `Privileged Role Administrator` (AD Role)
+  - `Application Administrator` (AD Role)
+- `Owner` role at the subscription scope which is being integrated
 
-  **Roles**:
+Absence of the above privileges will result in Access related issues when trying to run the Terraform.
 
-  - Reader
-  - Key Vault Reader
-  - Storage Blob Data Reader
-  - Storage Account Key Operator Service Role
-  - Custom read-only access for required resources
+## What does the Terraform do?
 
-  **API permissions**:
+Terraform is going to create a new <u>**Service Principal**</u> corresponding to our Multi-Tenant App Registration and assign below Roles/Permissions/Policies to it at the scope of the Subscription which is being integrated. Uptycs requires these set of Roles/Permissions to be able to fetch the required information from your Tenant:
+It creates the following resources:
 
-  - Directory.Read.All
-  - User.Read.All
-  - Group.Read.All
-  - Application.Read.All
-  - OnPremisesPublishingProfiles.ReadWrite.All
-  - Organization.Read.All
-  - UserAuthenticationMethod.Read.All
-  - Policy.Read.All
+**Roles**:
 
-  **Policies**:
+- Reader
+- Key Vault Reader
+- Storage Blob Data Reader
+- Storage Account Key Operator Service Role
+- Custom read-only access for required resources
 
-  - Key Vault Access Policy (certificate_permissions : List, Get)
+**API permissions**:
 
-## Prerequisites
+- Directory.Read.All
+- User.Read.All
+- Group.Read.All
+- Application.Read.All
+- OnPremisesPublishingProfiles.ReadWrite.All
+- Organization.Read.All
+- UserAuthenticationMethod.Read.All
+- Policy.Read.All
 
-Ensure you have the following privileges before you execute the Terraform Script:
+**Policies**:
 
-* The user should have `owner` role in subscription to create resources
-* Administrative roles:
-  * Global administrator
+- Key Vault Access Policy (certificate_permissions : List, Get)
 
 ## Authentication
```

Comment:
specify the subscription you want to integrate.

Reply:
specify subscription where?